src/yoroi-wallets/cardano/ShelleyWallet.ts

```diff
@@ -127,20 +123,24 @@ export class ShelleyWallet implements WalletInterface {
   rewardAddressHex: null | string = null
   version: undefined | string
   checksum: undefined | CardanoTypes.WalletChecksum
+  private _utxos: RawUtxo[]
 
   // =================== create =================== //
-  constructor(storage: typeof storageLegacy, networkId: NetworkId, id: string) {
+  constructor(storage: typeof storageLegacy, networkId: NetworkId, id: string, utxoManager: UtxoManager) {
     this.id = id
     this.storage = storage
     this.networkId = networkId
     this.defaultAsset = getDefaultAssetByNetworkId(this.networkId)
 
-    const config = this.getBackendConfig()
-    this.utxoStorage = makeUtxoStorage(this.storage, `/wallet/${this.id}/utxos`)
-    this.utxoService = initUtxo(this.utxoStorage, `${config.API_ROOT}/`)
+    this.utxoManager = utxoManager
+    this._utxos = utxoManager.initialUtxos
     this.encryptedStorage = makeWalletEncryptedStorage(id)
   }
 
+  get utxos() {
+    return [...this._utxos]
```

Comment:
shallow copy

Reply:
why a copy?

Reply:
yap, it looks unnecessary, since it's not mutating, always replacing it